django_project/stakeholder/views.py

```diff
@@ -68,3 +86,395 @@ def get_context_data(self, **kwargs):
+                    id=int(element)
+                )
+
+        return reminder
+    except Exception as e:
+        print(str(e))
+        return []
+
+
+def paginate(*args):
+    rows = args[0]
+    rows_per_page = args[1]
+    page = args[2]
+    paginator = Paginator(rows, rows_per_page)
+
+    try:
+        paginated_rows = paginator.page(page)
+    except PageNotAnInteger:
+        paginated_rows = paginator.page(1)
+    except EmptyPage:
+        paginated_rows = paginator.page(paginator.num_pages)
+
+    return paginated_rows
+
+
+def get_organisation_reminders(request):
+
+    reminders = Reminders.objects.filter(
+        user=request.user,
+        organisation=request.session[CURRENT_ORGANISATION_ID_KEY]
+    )
+
+    return reminders
+
+
+class RemindersView(DetailView):
+    template_name = 'reminders.html'
+    model = get_user_model()
+    slug_field = 'username'
+
+    def get_reminders(self, request, **kwargs):
+
+        reminders = get_organisation_reminders(request)
+
+        new_reminders = create_return_results(reminders)
+
+        reminders_page = self.request.GET.get('reminders_page', 1)
+
+        # Get the rows per page value from the query parameters
+        rows_per_page = self.request.GET.get('reminders_per_page', 5)
+
+        # paginate results
+        paginated_rows = paginate(
+            new_reminders,
+            rows_per_page,
+            reminders_page
+        )
+
+        return paginated_rows
+
+
+    # add the reminder aswell as schedule the task
+
+
+    def add_reminder(self, request):
+        if request.method == 'POST':
+            title = request.POST.get('title')
+            reminder_note = request.POST.get('reminder')
+            adjusted_datetime = self.adjust_timezone(request)
+
+            if request.POST.get('reminder_type') == 'personal':
+                reminder_type = Reminders.PERSONAL
+            else:
+                reminder_type = Reminders.EVERYONE
+            try:
+                organisation = Organisation.objects.get(
+                    id=self.request.session[CURRENT_ORGANISATION_ID_KEY]
+                )
+                # Save the reminder to the database
+                reminder = Reminders.objects.create(
+                    user=request.user,
+                    reminder = reminder_note,
+                    title=title,
+                    date=adjusted_datetime,
+                    type = reminder_type,
+                    organisation=organisation
+                )
+                # Schedule the Celery task to send the reminder email
+                task = send_reminder_email.apply_async(
+                    args=[reminder.id],
+                    eta=adjusted_datetime
+                )
+                task_id = task.id
+                reminder.task_id = task_id
+                reminder.save()
+
+                reminders = get_organisation_reminders(request)
+                results = create_return_results(reminders)
```

Comment:
use serializer instead

Reply:
the methods that are using this function will serialize it but when the object is being return as json some fields are not json serializable like the datatime field and other objects like user and organisation so this method is changing them into string represantions before the object is sent over the json response . is this okay?